assembly/src/ast/mod.rs

```diff
@@ -438,11 +472,14 @@ impl Deserializable for ModuleAst {
         } else {
             None
         };
-
+        let num_imports = source.read_u16()? as usize;
+        let imports_vec: Vec<LibraryPath> = Deserializable::read_batch_from(source, num_imports)?;
+        let imports = imports_vec.into_iter().map(|i| (i.last().to_string(), i)).collect();
```

Comment:
@jjcnn can you please elaborate a bit? My understanding is that `into_iter()` would take ownership and also avoid cloning? Is that incorrect?

Reply:
Yes, my understanding is also that using `into_iter()` would not result in any extra cloning.

Reply:
Ah, `into_iter` rather than `iter()`. Got it.

There's still two traversals when one would suffice, but I guess the Rust compiler can figure that out and optimize it.